testutil/checkers.go

```diff
@@ -23,13 +23,46 @@ import (
 	"bytes"
 	"fmt"
 	"io/ioutil"
+	"os"
 	"reflect"
 	"regexp"
 	"strings"
 
 	"gopkg.in/check.v1"
 )
 
+type filePresenceChecker struct {
+	*check.CheckerInfo
+	present bool
+}
+
+// FilePresent verifies that the given file exists.
+var FilePresent check.Checker = &filePresenceChecker{
+	CheckerInfo: &check.CheckerInfo{Name: "FilePresent", Params: []string{"filename"}},
+	present:     true,
+}
+
+// FileAbsent verifies that the given file does not exist.
+var FileAbsent check.Checker = &filePresenceChecker{
+	CheckerInfo: &check.CheckerInfo{Name: "FileAbsent", Params: []string{"filename"}},
+	present:     false,
+}
+
+func (c *filePresenceChecker) Check(params []interface{}, names []string) (result bool, error string) {
+	filename, ok := params[0].(string)
```

Comment:
Is this guaranteed to always have non-0 length?

Reply:
The checker frameworks seems to do that, yes.